crates/bevy_math/src/bounding/bounded2d/mod.rs

```diff
@@ -129,10 +139,32 @@ impl BoundingVolume for Aabb2d {
     }
 }
 
+impl IntersectsVolume<Self> for Aabb2d {
+    #[inline(always)]
+    fn intersects(&self, other: &Self) -> bool {
+        let x_overlaps = self.min.x <= other.max.x && self.max.x >= other.min.x;
+        let y_overlaps = self.min.y <= other.max.y && self.max.y >= other.min.y;
+        x_overlaps && y_overlaps
+    }
+}
+
+impl IntersectsVolume<BoundingCircle> for Aabb2d {
+    #[inline(always)]
+    fn intersects(&self, circle: &BoundingCircle) -> bool {
+        let closest_point = self.closest_point(circle.center);
+        let distance_squared = circle.center.distance_squared(closest_point);
+        let radius_squared = circle.radius().powi(2);
+        distance_squared < radius_squared
```

Comment:
This is nitpicking at edge cases, but I'd say this should be less than or equals, to account for the points at the surface of the ball.
```suggestion
        distance_squared <= radius_squared
```